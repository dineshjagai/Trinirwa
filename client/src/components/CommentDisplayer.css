.comment_cont {
    position: relative;
    width: 60%;
}

#comment_picture {
    width: 35px;
    height: 35px;
    border-radius: 35px
}
.comment_header {
    display: flex;
    flex-direction: row;
}

#editBtn {
    color: #0C8367;
    background: transparent;
    font-size: small;
}

.header_profile {
    margin: 4px 2px 4px 4px ;
}
.header_name {
    margin: 4px 4px 4px 2px ;
}
#cmnt_username {
    font-size: small;
    font-weight: bold;
    color: #0C8367;
 }

.cmnt_cnt_box {
    background-color: white;
    border-radius: 8px;
    position: absolute;
    left: 12%;
    top: 40%;
}

#cmnt_cnt {
    margin: 15px;
}

#editBtn:hover {
    text-decoration: underline;
    font-weight: bold ;
}