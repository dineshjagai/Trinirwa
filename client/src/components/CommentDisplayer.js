import React, { useEffect, useState } from 'react';
import './CommentDisplayer.css';
import DialogComment from './DialogComment';
import { getCurrentUsername } from '../auth/authServices';
import { getAvatar, updateComment } from './Module';

export default function CommentDisplayer({ data, handleDeleteComment }) {
  const [avatar, setAvatar] = useState('');
  const [content, setContent] = useState(data.content);
  const user = getCurrentUsername();
  const isOwner = data.user === user;
  const getData = async () => {
    await getAvatar(data.user).then((res) => {
      setAvatar(res.data.avatar[0].profile_picture);
    }).catch((err) => {
      console.log(err.message);
    });
  };

  const updatePost = (newContent) => {
    if (newContent === '') {
      return;
    }
    setContent(newContent);
    updateComment(data.comm_id, newContent).then((res) => {
      console.log(res.message);
    }).catch((err) => console.log(err.message));
  };

  useEffect(() => {
    getData();
  }, []);

  return (
    <div className="comment_cont" stlye={{ height: '200px', width: '100px', backgroundColor: 'red' }}>
      <div className="comment_header">
        <div className="header_profile">
          <img id="comment_picture" src={`/viewFile/${avatar}`} alt="" />
        </div>
        <div className="header_name">
          <span id="cmnt_username">{data.user}</span>
        </div>
      </div>
      <div className="cmnt_cntBtn_box" style={{ width: 'fit-content' }}>
        <div className="cmnt_cnt_box">
          <span id="cmnt_cnt">{content}</span>
        </div>
        <div className="cmnt_btns_box">
          <div style={{ display: isOwner ? 'block' : 'none' }} className="cmnt_EditBtn_box">
            <DialogComment id={data.comm_id} updatePost={updatePost} content={content} />
          </div>
          <div style={{ display: isOwner ? 'block' : 'none' }} className="cmnt_EditBtn_box">
            <button onClick={() => handleDeleteComment(data.comm_id)} id="deleteBtn" type="button">Delete</button>
          </div>
        </div>
      </div>

    </div>
  );
}
