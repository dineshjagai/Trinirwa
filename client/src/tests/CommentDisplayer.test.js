import React from 'react';
import renderer from 'react-test-renderer';
import CommentDisplayer from '../components/CommentDisplayer';

test('Test CommentDisplayer snapshot', () => {
  const d = {
    'user' : 'isimbib',
    'tweet_date' :  '2008-09-07',
    'content' : 'Hiii',
    'tweet_likes': 0,
    'tweet_id' :  '1',
    'type' : 'picture',
    'tweet_blocks': 0,
    'profile_picture': 'logo512.png',
    'comm_id': 3,
  };

  const handle = (id) => {
    if (id) {
      console.log("owner");
    } else {
      console.log("HERE {id}");
    }
  };
  const component = renderer.create(<CommentDisplayer data={d} handleDeleteComment={handle}/>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
