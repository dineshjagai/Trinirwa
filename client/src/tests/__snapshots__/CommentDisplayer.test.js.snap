// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CommentDisplayer snapshot 1`] = `
<div
  className="comment_cont"
  stlye={
    Object {
      "backgroundColor": "red",
      "height": "200px",
      "width": "100px",
    }
  }
>
  <div
    className="comment_header"
  >
    <div
      className="header_profile"
    >
      <img
        alt=""
        id="comment_picture"
        src="/api/viewFile/"
      />
    </div>
    <div
      className="header_name"
    >
      <span
        id="cmnt_username"
      >
        isimbib
      </span>
    </div>
  </div>
  <div
    className="cmnt_cntBtn_box"
    style={
      Object {
        "width": "fit-content",
      }
    }
  >
    <div
      className="cmnt_cnt_box"
    >
      <span
        id="cmnt_cnt"
      >
        Hiii
      </span>
    </div>
    <div
      className="cmnt_btns_box"
    >
      <div
        className="cmnt_EditBtn_box"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <div>
          <button
            id="editBtn"
            onClick={[Function]}
            type="button"
          >
            Edit
          </button>
        </div>
      </div>
      <div
        className="cmnt_EditBtn_box"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <button
          id="deleteBtn"
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
